src/searchbar.cpp

```diff
@@ -183,20 +182,17 @@ void SearchBarLineEdit::updateCompletion()
 
 void SearchBarLineEdit::openCompletion(const QModelIndex &index)
 {
-    if (m_urlList.size() != 0) {
-        openCompletion(index.data().toString(), index.row());
-    }
-}
-
-void SearchBarLineEdit::openCompletion(const QString& text, int index)
-{
-    QUrl url;
-    if (this->text().compare(text, Qt::CaseInsensitive) == 0) {
-        url = m_urlList.at(index);
-    } else {
-        url = m_urlList.last();
+    if (m_suggestionModel.rowCount() != 0) 
+    {
+        QUrl url;
+        auto editText = index.data().toString();
+        if (this->text().compare(editText, Qt::CaseInsensitive) == 0) {
+            url = index.data(Qt::UserRole).toUrl();
+        } else {
+            url = m_suggestionModel.lastIndex().data(Qt::UserRole).toUrl();
+        }
```

Comment:
I guess the logic here is select the first suggestion if it fully matches the search text otherwise perform full text search which should be at the end. However if the ZIM file doesn't contain a full text search database then the full text search suggestion is not generated.

Reply:
Done, added clause to simply return if we do not have full text in this situation.